### DISCUSSION TITLE
Better Discord memory leak

### DISCUSSION THREAD

**Discussion by Serega007RU** [tsukasa/BdBrowser#17] [Problems and issues]
Hello! Faced with the problem of high consumption of RAM discord. I understand that the problem is most likely not with BDBrowser, but can you tell me how this problem can be solved and in general where to look?

In this screenshot you can see the Google Chrome task Manager, which shows that the discord tab consumes more than 5 gigabytes of RAM (sorry that everything is in Russian)
![image](https://user-images.githubusercontent.com/64780091/210589122-0c72f1c2-b566-49df-a090-287d8eb763b9.png)
Such a large amount of memory consumption is not achieved immediately, but only with prolonged use of the discord (approximately more than 6 hours). If you use the discord even longer, sooner or later it will crash altogether. Unfortunately, I did not take a screenshot of where the discord is crashing, but there was an error something like with too high memory consumption, when I catch this error again, I will take a screenshot of this error.

**Comment by tsukasa** [comment]
Greetings,

this is tough for me to answer, I do not experience memory leaks with Discord after BetterDiscord pushed a few fixes with the last updates. It generally jumps between 400MB (idle) and 600MB (when switching between servers) on my end and this is with DevilBro's plugins active.

I would try the following:
- Log in from a private/incognito tab with BdBrowser enabled in incognito mode so that only BetterDiscord loads, without any themes and plugins. Observe if the same behaviour occurs.
- If no memory leak occurs and the RAM usage stays stable, start disabling plugins in your main tab and see if memory continues to leak there.
- In case no memory leak occurs, re-enable a few plugins and observe. Rinse and repeat until you have hit a batch of plugins that cause the memory leak.

To double-check whether BetterDiscord has an impact on the occurrence of the leaks, you can also disable BdBrowser for private mode, so that opening a private Discord tab loads vanilla Discord only. This makes it easier to follow the memory consumption of that specific tab.

The Memory tab in the developer tools with the "Allocation sampling" mode (last option) can also be used to run longer memory analysis. Start the profiler, wait a few hours, stop the profiler and see which calls have been the most demanding. Note that the names might be mangled a bit, however with a bit of context you might be able to ascertain certain memory hogs (or at least their source).

I can imagine that BdBrowser does contribute to some excess CPU/memory usage. If you want to disable some potential memory hogs in BdBrowser, consider removing the `#loadingObserver` in `ReplaceLoadingAnimation` (`backend/src/modules/loadingScreen.js`) and the `addVersionInfoObserver` in `addExtensionVersionInfo` (`frontent/src/modules/runtimeInfo.js`).

**Comment by Serega007RU** [comment]
Well, I'll try, but it's quite difficult due to the fact that the memory leak pops up only for a long time (more than 3 hours)

**Comment by ClaraCrazy** [comment]
Throwing in my 2 cents. I just checked (having my chromium fork running 24/7 basically), and with my ~60 plugins + custom theme I'm at around 1.3GB. So that number really does seem absurdly high.

We don't talk about all my plugins, lol
